### PR TITLE
Refactoring of polygon editors to a common code base (v2)

### PR DESCRIPTION
**Pull request by ghost**
Follow up of #10768.

This time more compact, cleaner and focused on just deduplicating the polygon editor code. No API changes.

### CODE REVIEW

**Comment by akien-mga** [comment]
This will need a rebase to solve recent merge conflicts, but before spending too much time rebase every other day, the current commits should be reviewed to confirm that it's the approach we want.

**Comment by groud** [comment]
Ok I have to take a look at this.

File: editor/plugins/abstract_polygon_2d_editor.h [hunk 1]
```diff
@@ -0,0 +1,131 @@
+/*************************************************************************/
+/*  abstract_polygon_2d_editor.cpp                                       */
```
Comment (Noshyaar) [inline editor/plugins/abstract_polygon_2d_editor.h]: `.h` ?

**Comment by groud** [comment]
I think the work is globally good. The function naming may be improved but I like the idea anyway.
I think it deserves a rebase.

File: editor/plugins/abstract_polygon_2d_editor.h (hunk 1 above)
Comment (poke1024) [reply editor/plugins/abstract_polygon_2d_editor.h]: Fixed this in the rebase below.

**Comment by groud** [comment]
Ok sorry for beging late, we discussed this PR in the PR monday.
Will merge as soon as it is rebased. :)

**Comment by poke1024** [comment]
@groud rebased now.

**Comment by groud** [comment]
Hum. I think this PR also needs a squash to make it a single commit. Please read how to make a squash here: http://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html

**Comment by poke1024** [comment]
@groud rebased. Is the authoring fixed now?

**Comment by groud** [comment]
> @groud rebased. Is the authoring fixed now?

Perfect. I forgot to merge this one... Merging then ;)

**Pull request merged**